packages/smart-contracts/src/contracts/ERC20EscrowToPayV1.sol

```diff
@@ -0,0 +1,385 @@
+        
+        delete invoiceMapping[_paymentRef];
+
+        emit EscrowCompleted(_paymentRef);  
+    }
+    
+    /**
+     * @notice Allaws the payer to initiate a dispute if they have an isssue with the payee.
+     * @param _paymentRef Reference of the Invoice related.
+     */
+    function openDispute(bytes memory _paymentRef) external {
+        require(msg.sender == invoiceMapping[_paymentRef].payer, 
+            "ERC20EscrowToPay: You are not the valid payer. Only the valid payer can open a dispute!");
+        require(invoiceMapping[_paymentRef].amount != 0, 
+            "ERC20EscrowToPay: No dispute found, Wrong payment reference?");
+        require(disputeMapping[_paymentRef].amount == 0, "ERC20EscrowToPay: A dispute already exists, is this the correct paymentRef?");
+
+        disputeMapping[_paymentRef] = Dispute(
+            _paymentRef,
+            invoiceMapping[_paymentRef].paymentToken, 
+            invoiceMapping[_paymentRef].amount,
+            invoiceMapping[_paymentRef].payee,
+            invoiceMapping[_paymentRef].payer,
+            invoiceMapping[_paymentRef].feeAmount,
+            invoiceMapping[_paymentRef].feeAddress,
+            0,
+            0,
+            TokenTimelock(0x0000000000000000000000000000000000000000)
+        );
+        delete invoiceMapping[_paymentRef];
+
+        emit DisputeOpened(_paymentRef);
+    }
+
+    /** 
+     * @notice Closes an open dispute and pays the invoice request to it's issuer.
+     * @param _paymentRef Reference of the related Invoice.
+     * @dev The fees is paid to the fee address.
+     */
+    function resolveDispute(bytes memory _paymentRef) external OnlyPayer(_paymentRef) {
+        require(msg.sender == disputeMapping[_paymentRef].payer,
+            "ERC20EscrowToPayV1: Only the payer can resolve an active dispute."
+        );
+        require(disputeMapping[_paymentRef].amount != 0, 
+            "ERC20EscrowToPayV1: No open dispute found, wrong payment reference?"
+        );
+        require(_withdraw( _paymentRef, disputeMapping[_paymentRef].payee), 
+            "ResolveDispute: call to _withdraw() Failed"
+        ); 
+        delete disputeMapping[_paymentRef];
+        
+        emit DisputeResolved(_paymentRef);
+    }
+
+    /**
+     * @notice Creates a tokentimelock contract and returns the escrowed funds to the payer after 12 months.
+     * @param _paymentRef Reference of the Invoice related.
+     * @dev  The fees is paid to the fee address, disputed funds are locked in tokentimelock contract ith the payer as _beneficiary.
+     * @return tokentimelock The TokenTimelock contract address.
+     */
+    function lockDisputedFunds(bytes memory _paymentRef) external OnlyPayer(_paymentRef) returns (address tokentimelock) {
+        require(disputeMapping[_paymentRef].amount != 0, "ERC20EscrowToPay: No dispute found, wrong paymentRef?");
```

Comment:
I don't understand the need for the intermediate step "Dispute". It means that to lock funds, the payer will need 2 transactions, why ?